### ISSUE TITLE
Hierarchical clustering needs clean up.

### ISSUE THREAD

**Issue by rmcgibbo** [msmbuilder/msmbuilder#366]
Currently, we have a wrapper for `sklearn.clustering.Ward` and `LandmarkAgglomerative`. There are a number of problems.

 - The Ward class in sklearn is deprecated.
 - Neither of these classes use libdistance, which means they don't handle rmsd using the same pattern as KCenters, KMedoids, etc.
 - None of the models are exposed on the command line

We should make sure all of these classes have a transform method to assign labels for new datasets, even if the method for computing these labels is arguably not optimal. Without this capability, with GMRQ, the models are basically useless. `LandmarkAgglomerative` does this somewhat reasonably, IMO, using `POOLING_FUNCTIONS`, for `average`, `complete` and `single` linkage, but I'm not convinced we can't to something reasonable for Ward. Even, I think, using the `average` linkage pooling function with clusters that were generated using `Ward` would be reasonable, I think.

Also, the ability to assign labels to new datasets is _essential_ for production calculations on non-toy datasets, since all of these require O(N^2) memory. Naturally, this means that we're going to be doing the clustering on either a strided or randomly-sampled subset of the data, so then we're really going to want to assign state labels for the full dataset.

IMO, we should probably use `fastcluster` / `scipy.cluster.hierarchy`, and skip using `sklearn.cluster.AgglomerativeClustering`

**Comment by mpharrigan** [comment]
@brookehus did you do this? Can this be closed?

**Comment by brookehus** [comment]
Yup, all three bullet points have been addressed in #874.
